src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -1561,12 +1594,14 @@ address MacroAssembler::target_addr_for_insn(address insn_addr) {
     offset = get_offset_of_conditional_branch(insn_addr);
   } else if (NativeInstruction::is_pc_relative_at(insn_addr)) {      // auipc, addi/jalr/load
     offset = get_offset_of_pc_relative(insn_addr);
-  } else if (NativeInstruction::is_movptr_at(insn_addr)) {           // movptr
+  } else if (NativeInstruction::is_movptr1_at(insn_addr)) {          // movptr
     return get_target_of_movptr(insn_addr);
   } else if (NativeInstruction::is_li64_at(insn_addr)) {             // li64
     return get_target_of_li64(insn_addr);
   } else if (NativeInstruction::is_li32_at(insn_addr)) {             // li32
     return get_target_of_li32(insn_addr);
+  } else if (NativeInstruction::is_movptr2_at(insn_addr)) {          // movptr2
```

Comment:
You could move that `else if` block right under the `NativeInstruction::is_movptr1_at` one.